packages/cli/src/__tests__/credentials-overwrites.test.ts

```diff
@@ -0,0 +1,134 @@
+import type { GlobalConfig } from '@n8n/config';
+import { mock } from 'jest-mock-extended';
+import { UnrecognizedCredentialTypeError, type Logger } from 'n8n-core';
+import type { ICredentialType } from 'n8n-workflow';
+
+import type { CredentialTypes } from '@/credential-types';
+import { CredentialsOverwrites } from '@/credentials-overwrites';
+
+describe('CredentialsOverwrites', () => {
+	const testCredentialType = mock<ICredentialType>({ name: 'test', extends: ['parent'] });
+	const parentCredentialType = mock<ICredentialType>({ name: 'parent', extends: undefined });
```

Comment:
Should these also have `properties` that define `username` and `password`?

Reply:
since this code doesn't use `properties` for anything, I think we can skip them on these mocks.